### PR TITLE
feat(inference): add reusable optimizer loop with optax

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add reusable optimizer loop (optimize_params) for inference workflows
- support optional constrained-parameter transforms during optimization
- return structured optimization traces and best-parameter snapshots
- add unit tests for loss reduction and bounded-transform optimization

## Stack position
PR 4 / 8 in the gradient-production plan

### CODE REVIEW

File: rubix/inference/optimize.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        updates, opt_state = optimizer.update(grads, opt_state, trainable_params)
+        trainable_params = optax.apply_updates(trainable_params, updates)
+
+        grad_norm = optax.global_norm(grads)
+        update_norm = optax.global_norm(updates)
+
+        loss_value = float(value)
+        loss_history.append(loss_value)
+        grad_norm_history.append(float(grad_norm))
+
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/optimize.py]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/199#discussion_r3100920485)

File: rubix/inference/optimize.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    """Return a mutable dictionary copy from a nested parameter tree."""
+    return {component: dict(fields) for component, fields in tree.items()}
+
+
+def optimize_params(
+    pipeline: Any,
+    params_init: ParamsTree,
+    static_data: Any,
+    target: jnp.ndarray,
+    learning_rate: float = 1e-3,
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/optimize.py:40]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/199#discussion_r3100920508)

[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/optimize.py:40]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/199#discussion_r3100920508)
